doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+        /// and a local source (Installing, Installed) together  
+        /// IMPLEMENTATION NOTE: Windows Package Manager currently only supports 
+        /// one local source and one remote source per composite catalog.
+        /// DESIGN NOTE: It seems likely that in the future we will want to support 
+        /// letting callers use two local sources (i.e. both Installing and Installed)
+        IVector<AppCatalog> Catalogs { get; };
+        /// The local source(s)
+        LocalAppCatalog LocalAppCatalog;
+        /// Sets the default search behavior if the catalog is a composite catalog.
+        CompositeSearchBehavior CompositeSearchBehavior;
+    }
+
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppInstallScope
+    {
+        User,
+        Machine,
+    };
+
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppInstallMode
+    {
+        /// The default experience for the installer. Installer may show some UI.
+        Default,
+        /// Runs the installer in silent mode. This suppresses the installer's UI to the extent 
+        /// possible (installer may still show some required UI).
+        Silent,
+        /// Runs the installer in interactive mode.
+        Interactive,
+    };
+
+    /// Options when installing a package.
+    /// Intended to allow full compatibility with the "winget install" command line interface.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass InstallOptions
+    {
+        InstallOptions();
+
+        /// DESIGN NOTE: --Exact from the winget install command line is implied here by virtue of the find 
+        /// then install object semantics.
+        /// Likewise --Query from the command line is not implemented here since the query is required up 
+        /// front in order to get the CatalogPackage object.
+        /// DESIGN NOTE: Using PackageVersionInfo rather than CatalogPackage felt like it would be too 
+        /// specific here. CatalogPackage and PackageVersionId (or if unspecified then the latest one) may be nicer
+        /// if we end up having versions that have different applicability or other more complex logic about 
+        /// which version is preferred.
+        CatalogPackage CatalogPackage;
+        /// Optionally specifies the version from the package to install. If unspecified the version matching 
+        /// CatalogPackage.GetLatestVersion() is used.
+        PackageVersionId PackageVersionId;
+
+        /// Specifies alternate location to install package (if supported).
+        String PreferredInstallLocation;
+        /// User or Machine. 
+        AppInstallScope AppInstallScope;
+        /// Silent, Interactive, or Default
+        AppInstallMode AppInstallMode;
+        /// Directs the logging to a log file. Callers must provide a path to a file that the Windows Package 
+        /// Manager package has write access to.

```

Comment:
Can I leave this empty if I don't want a log file?

Must the file already exist?

What are the minimum ACLs required to ensure that the Package Manager package can write to the file?

Reply:
Yes, can be empty. Package manager is a full trust package, so the user it's running as has to have access.

Reply:
If it can be empty, then comment should not say "Callers must provide..."

Reply:
Good point.

Reply:
RECOMMEND: Document requirements of log path.  Fix comment as well.

CONSIDER: Review the kinds and locations of paths you support.